statsmodels/discrete/discrete_model.py

```diff
@@ -2525,25 +2528,28 @@ def _score_nbin(self, params, Q=0):
         y = self.endog[:,None]
         mu = self.predict(params)[:,None]
         a1 = 1/alpha * mu**Q
+
+        prob = a1 / (a1+mu)
         if Q: # nb1
-            dparams = exog*mu/alpha*(np.log(1/(alpha + 1)) +
+            # Note: 1/(alpha+1) == (mu/alpha) / (mu+mu/alpha) == a1/(mu+a1)
+            dparams = exog * a1 * (np.log(prob) +
                        special.digamma(y + mu/alpha) -
                        special.digamma(mu/alpha))
-            dalpha = ((alpha*(y - mu*np.log(1/(alpha + 1)) -
+            dalpha = ((alpha*(y - mu*np.log(prob) -
                               mu*(special.digamma(y + mu/alpha) -
                               special.digamma(mu/alpha) + 1)) -
-                       mu*(np.log(1/(alpha + 1)) +
+                       mu*(np.log(prob) +
                            special.digamma(y + mu/alpha) -
                            special.digamma(mu/alpha)))/
                        (alpha**2*(alpha + 1))).sum()
 
         else: # nb2
-            dparams = exog*a1 * (y-mu)/(mu+a1)
+            dparams = exog * a1 * (y-mu)/(mu+a1)
             da1 = -alpha**-2
-            dalpha = (special.digamma(a1+y) - special.digamma(a1) + np.log(a1)
-                        - np.log(a1+mu) - (a1+y)/(a1+mu) + 1).sum()*da1
+            dalpha = (special.digamma(a1+y) - special.digamma(a1) + np.log(prob)
+                        - (y-mu)/(a1+mu)).sum()*da1
```

Comment:
There are two independent algebraic substitutions in the `dalpha = ...` line.

1) `np.log(a1) - np.log(a1+mu) == np.log(a1/(a1+mu)) == np.log(prob)`

2) `-(a1+y)/(a1+mu) + 1 == -(a1+y-a1-mu)/(a1+mu) == -(y-mu)/(a1+mu)`

This pair shows up again below.